migrations/20200515204647-initial-table-creation.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    /*
+      Add altering commands here.
+      Return a promise to correctly handle asynchronicity.
+
+      Example:
+      return queryInterface.createTable('users', { id: Sequelize.INTEGER });
+    */
+    await queryInterface.createTable('authors', {
+      id: { type: Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
+      nameFirst: { type: Sequelize.STRING, allowNull: false },
+      nameLast: { type: Sequelize.STRING, allowNull: false },
+      createdAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP') },
+      updatedAt: {
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP')
+      },
+      deletedAt: { type: Sequelize.DATE },
+    })
+
+    await queryInterface.createTable('genres', {
+      id: { type: Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
+      name: { type: Sequelize.STRING, allowNull: false },
+      createdAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP') },
+      updatedAt: {
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP')
+      },
+      deletedAt: { type: Sequelize.DATE },
+    })
+
+    await queryInterface.createTable('novels', {
+      id: { type: Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
+      title: { type: Sequelize.STRING, allowNull: false },
+      authorID: { type: Sequelize.INTEGER, references: { model: 'authors', key: 'id' } },
+      createdAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP') },
+      updatedAt: {
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP')
+      },
+      deletedAt: { type: Sequelize.DATE },
+    })
+
+    return queryInterface.createTable('novelsGenres', {
+      genreId: { type: Sequelize.INTEGER, references: { model: 'genres', key: 'id' }, primaryKey: true },
+      novelId: { type: Sequelize.INTEGER, references: { model: 'novels', key: 'id' }, primaryKey: true },
+      createdAt: { type: Sequelize.DATE, defaultValue: Sequelize.literal('CURRENT_TIMESTAMP') },
+      updatedAt: {
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP')
+      },
+      deletedAt: { type: Sequelize.DATE },
+    })
+
+  },
+
+  down: async (queryInterface) => {
+    /*
+      Add reverting commands here.
+      Return a promise to correctly handle asynchronicity.
+
+      Example:
+      return queryInterface.dropTable('users');
+    */
+    await queryInterface.dropTable('authors')
+
+    await queryInterface.dropTable('genres')
+
+    await queryInterface.dropTable('novels')
+
+    return queryInterface.dropTable('novelsGenres')
```

Comment:
This would fail. You can't drop authors before novels and you can't drop novels or genres before novelsGenres because of foreign key constraints.